src/server/worker/workerrequests.js

```diff
@@ -1111,6 +1112,81 @@ function WorkerRequests(mainLogger, gmeConfig) {
             .nodeify(callback);
     }
 
+    /**
+     *
+     * @param webgmeToken
+     * @param parameters
+     * @param callback
+     */
+    function updateProjectFromFile(webgmeToken, parameters, callback) {
+        var jsonProject,
+            context,
+            storage,
+            blobClient = getBlobClient(webgmeToken);
+
+        getConnectedStorage(webgmeToken)
+            .then(function (storage_) {
+                storage = storage_;
+                return _getCoreAndRootNode(storage, parameters.projectId, parameters.commitHash, parameters.branchName);
+            })
+            .then(function (context_) {
+                context = context_;
+                return _importProjectPackage(blobClient, parameters.blobHash, true);
+            })
+            .then(function (jsonProject_) {
+                jsonProject = jsonProject_;
+
+                return storageUtils.insertProjectJson(context.project,
+                    jsonProject,
+                    {
+                        branch: parameters.branchName,
+                        parentCommit: [context.commitObject[STORAGE_CONSTANTS.MONGO_ID]],
+                        commitMessage: 'update project from file'
+                    });
+            })
+            .catch(function (err) {
+                logger.error('updateProjectFromFile failed with error', err);
+                throw err;
+            })
+            .nodeify(callback);
+    }
+
+    /**
+     *
+     * @param webgmeToken
+     * @param parameters
+     * @param callback
+     */
+    function squashCommits(webgmeToken, parameters, callback) {
```

Comment:
Do we really need a worker-request for this?
All work is done by the server in any case..